conjure-core/src/main/java/com/palantir/conjure/defs/validator/HttpPathValidator.java

```diff
@@ -37,7 +37,7 @@
     private HttpPathValidator() {}
 
     public static final String PATTERN = "[a-z][a-z0-9]*([A-Z0-9][a-z0-9]+)*";
-    private static final Pattern SEGMENT_PATTERN = Pattern.compile("^[a-zA-Z][a-zA-Z0-9_-]*$");
+    private static final Pattern SEGMENT_PATTERN = Pattern.compile("^[a-zA-Z][a-zA-Z0-9._-]*$");
```

Comment:
@iamdanfox slightly confused as i see this block duplicated in both this file as well as com/palantir/conjure/parser/services/PathString.java

should i be making the change to both?